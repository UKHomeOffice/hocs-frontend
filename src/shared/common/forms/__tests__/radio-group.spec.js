import React from 'react';
import RadioGroup from '../radio-group.jsx';

const choices = [
    {label: 'isA', value: 'A'},
    {label: 'isB', value: 'B'},
    {label: 'isC', value: 'C'}
];

describe('Form radio group component', () => {
    it('should render with default props', () => {
        expect(
            render(<RadioGroup name="radio-group" choices={choices}/>)
        ).toMatchSnapshot();
    });
    it('should render with value when passed', () => {
        expect(
            render(<RadioGroup name="radio-group" choices={choices} value="A"/>)
        ).toMatchSnapshot();
    });
    it('should render with label when passed', () => {
        expect(
            render(<RadioGroup name="radio-group" choices={choices} label="My text field"/>)
        ).toMatchSnapshot();
    });
    it('should render with hint when passed', () => {
        expect(
            render(<RadioGroup name="radio-group" choices={choices} hint="Put some text in the box below"/>)
        ).toMatchSnapshot();
    });
    it('should render with error when passed', () => {
        expect(
            render(<RadioGroup name="radio-group" choices={choices} error="Some error message"/>)
        ).toMatchSnapshot();
    });
    it('should render disabled when passed', () => {
        expect(
            render(<RadioGroup name="radio-group" choices={choices} disabled={true}/>)
        ).toMatchSnapshot();
    });
    it('should execute callback on initialization', () => {
        const mockCallback = jest.fn();
        const wrapper = shallow(
            <RadioGroup name="radio-group" choices={choices} updateState={mockCallback}/>
        );
        expect(mockCallback).toHaveBeenCalledTimes(1);
        expect(mockCallback).toHaveBeenCalledWith({'radio-group': 'A'});
    });
    it('should execute callback on change', () => {
        const mockCallback = jest.fn();
        let firstValue = 'A';
        let secondValue = 'B';
        const wrapper = shallow(
            <RadioGroup name="radio-group" choices={choices} updateState={mockCallback}/>
        );

        mockCallback.mockReset();

        wrapper.find(`#radio-group-${firstValue}`).simulate('change', {target: {value: firstValue}});
        expect(mockCallback).toHaveBeenCalledTimes(1);
        expect(mockCallback).toHaveBeenCalledWith({'radio-group': firstValue});

        wrapper.find(`#radio-group-${secondValue}`).simulate('change', {target: {value: secondValue}});
        expect(mockCallback).toHaveBeenCalledTimes(2);
        expect(mockCallback).toHaveBeenCalledWith({'radio-group': secondValue});
    });
});

