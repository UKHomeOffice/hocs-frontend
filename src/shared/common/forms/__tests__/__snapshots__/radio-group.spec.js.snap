// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form radio group component should render disabled when passed 1`] = `
<div
  class="govuk-form-group"
>
  <fieldset
    class="govuk-fieldset "
    disabled=""
    id="radio-group"
  >
    <legend
      class="govuk-fieldset__legend"
      id="radio-group-legend"
    >
      <span
        class="govuk-fieldset__heading govuk-label--s"
      />
    </legend>
    <div
      class="govuk-radios"
      id="radio-group-radios"
    >
      <div
        class="govuk-radios__item"
      >
        <input
          class="govuk-radios__input"
          id="radio-group-A"
          name="radio-group"
          type="radio"
          value="A"
        />
        <label
          class="govuk-label govuk-radios__label"
          for="radio-group-A"
        >
          isA
        </label>
      </div>
      <div
        class="govuk-radios__item"
      >
        <input
          class="govuk-radios__input"
          id="radio-group-B"
          name="radio-group"
          type="radio"
          value="B"
        />
        <label
          class="govuk-label govuk-radios__label"
          for="radio-group-B"
        >
          isB
        </label>
      </div>
      <div
        class="govuk-radios__item"
      >
        <input
          class="govuk-radios__input"
          id="radio-group-C"
          name="radio-group"
          type="radio"
          value="C"
        />
        <label
          class="govuk-label govuk-radios__label"
          for="radio-group-C"
        >
          isC
        </label>
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Form radio group component should render with default props 1`] = `
<div
  class="govuk-form-group"
>
  <fieldset
    class="govuk-fieldset "
    id="radio-group"
  >
    <legend
      class="govuk-fieldset__legend"
      id="radio-group-legend"
    >
      <span
        class="govuk-fieldset__heading govuk-label--s"
      />
    </legend>
    <div
      class="govuk-radios"
      id="radio-group-radios"
    >
      <div
        class="govuk-radios__item"
      >
        <input
          class="govuk-radios__input"
          id="radio-group-A"
          name="radio-group"
          type="radio"
          value="A"
        />
        <label
          class="govuk-label govuk-radios__label"
          for="radio-group-A"
        >
          isA
        </label>
      </div>
      <div
        class="govuk-radios__item"
      >
        <input
          class="govuk-radios__input"
          id="radio-group-B"
          name="radio-group"
          type="radio"
          value="B"
        />
        <label
          class="govuk-label govuk-radios__label"
          for="radio-group-B"
        >
          isB
        </label>
      </div>
      <div
        class="govuk-radios__item"
      >
        <input
          class="govuk-radios__input"
          id="radio-group-C"
          name="radio-group"
          type="radio"
          value="C"
        />
        <label
          class="govuk-label govuk-radios__label"
          for="radio-group-C"
        >
          isC
        </label>
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Form radio group component should render with error when passed 1`] = `
<div
  class="govuk-form-group govuk-form-group--error"
>
  <fieldset
    class="govuk-fieldset "
    id="radio-group"
  >
    <legend
      class="govuk-fieldset__legend"
      id="radio-group-legend"
    >
      <span
        class="govuk-fieldset__heading govuk-label--s"
      />
    </legend>
    <span
      class="govuk-error-message"
      id="radio-group-error"
    >
      Some error message
    </span>
    <div
      class="govuk-radios"
      id="radio-group-radios"
    >
      <div
        class="govuk-radios__item"
      >
        <input
          class="govuk-radios__input"
          id="radio-group-A"
          name="radio-group"
          type="radio"
          value="A"
        />
        <label
          class="govuk-label govuk-radios__label"
          for="radio-group-A"
        >
          isA
        </label>
      </div>
      <div
        class="govuk-radios__item"
      >
        <input
          class="govuk-radios__input"
          id="radio-group-B"
          name="radio-group"
          type="radio"
          value="B"
        />
        <label
          class="govuk-label govuk-radios__label"
          for="radio-group-B"
        >
          isB
        </label>
      </div>
      <div
        class="govuk-radios__item"
      >
        <input
          class="govuk-radios__input"
          id="radio-group-C"
          name="radio-group"
          type="radio"
          value="C"
        />
        <label
          class="govuk-label govuk-radios__label"
          for="radio-group-C"
        >
          isC
        </label>
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Form radio group component should render with hint when passed 1`] = `
<div
  class="govuk-form-group"
>
  <fieldset
    class="govuk-fieldset "
    id="radio-group"
  >
    <legend
      class="govuk-fieldset__legend"
      id="radio-group-legend"
    >
      <span
        class="govuk-fieldset__heading govuk-label--s"
      />
    </legend>
    <span
      class="govuk-hint"
    >
      Put some text in the box below
    </span>
    <div
      class="govuk-radios"
      id="radio-group-radios"
    >
      <div
        class="govuk-radios__item"
      >
        <input
          class="govuk-radios__input"
          id="radio-group-A"
          name="radio-group"
          type="radio"
          value="A"
        />
        <label
          class="govuk-label govuk-radios__label"
          for="radio-group-A"
        >
          isA
        </label>
      </div>
      <div
        class="govuk-radios__item"
      >
        <input
          class="govuk-radios__input"
          id="radio-group-B"
          name="radio-group"
          type="radio"
          value="B"
        />
        <label
          class="govuk-label govuk-radios__label"
          for="radio-group-B"
        >
          isB
        </label>
      </div>
      <div
        class="govuk-radios__item"
      >
        <input
          class="govuk-radios__input"
          id="radio-group-C"
          name="radio-group"
          type="radio"
          value="C"
        />
        <label
          class="govuk-label govuk-radios__label"
          for="radio-group-C"
        >
          isC
        </label>
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Form radio group component should render with label when passed 1`] = `
<div
  class="govuk-form-group"
>
  <fieldset
    class="govuk-fieldset "
    id="radio-group"
  >
    <legend
      class="govuk-fieldset__legend"
      id="radio-group-legend"
    >
      <span
        class="govuk-fieldset__heading govuk-label--s"
      >
        My text field
      </span>
    </legend>
    <div
      class="govuk-radios"
      id="radio-group-radios"
    >
      <div
        class="govuk-radios__item"
      >
        <input
          class="govuk-radios__input"
          id="radio-group-A"
          name="radio-group"
          type="radio"
          value="A"
        />
        <label
          class="govuk-label govuk-radios__label"
          for="radio-group-A"
        >
          isA
        </label>
      </div>
      <div
        class="govuk-radios__item"
      >
        <input
          class="govuk-radios__input"
          id="radio-group-B"
          name="radio-group"
          type="radio"
          value="B"
        />
        <label
          class="govuk-label govuk-radios__label"
          for="radio-group-B"
        >
          isB
        </label>
      </div>
      <div
        class="govuk-radios__item"
      >
        <input
          class="govuk-radios__input"
          id="radio-group-C"
          name="radio-group"
          type="radio"
          value="C"
        />
        <label
          class="govuk-label govuk-radios__label"
          for="radio-group-C"
        >
          isC
        </label>
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Form radio group component should render with value when passed 1`] = `
<div
  class="govuk-form-group"
>
  <fieldset
    class="govuk-fieldset "
    id="radio-group"
  >
    <legend
      class="govuk-fieldset__legend"
      id="radio-group-legend"
    >
      <span
        class="govuk-fieldset__heading govuk-label--s"
      />
    </legend>
    <div
      class="govuk-radios"
      id="radio-group-radios"
    >
      <div
        class="govuk-radios__item"
      >
        <input
          checked=""
          class="govuk-radios__input"
          id="radio-group-A"
          name="radio-group"
          type="radio"
          value="A"
        />
        <label
          class="govuk-label govuk-radios__label"
          for="radio-group-A"
        >
          isA
        </label>
      </div>
      <div
        class="govuk-radios__item"
      >
        <input
          class="govuk-radios__input"
          id="radio-group-B"
          name="radio-group"
          type="radio"
          value="B"
        />
        <label
          class="govuk-label govuk-radios__label"
          for="radio-group-B"
        >
          isB
        </label>
      </div>
      <div
        class="govuk-radios__item"
      >
        <input
          class="govuk-radios__input"
          id="radio-group-C"
          name="radio-group"
          type="radio"
          value="C"
        />
        <label
          class="govuk-label govuk-radios__label"
          for="radio-group-C"
        >
          isC
        </label>
      </div>
    </div>
  </fieldset>
</div>
`;
